sklearn/isotonic.py

```diff
@@ -343,8 +345,12 @@ def transform(self, T):
 
         T = check_array(T, dtype=dtype, ensure_2d=False)
 
-        if len(T.shape) != 1:
-            raise ValueError("Isotonic regression input should be a 1d array")
+        if not (len(T.shape) == 1 or (len(T.shape) == 2 and T.shape[1] == 1)):
+            msg = "Isotonic regression input should be a 1d array or " \
+                  "2d array with 1 feature"
+            raise ValueError(msg)
```

Comment:
instead of repeating this code, how about renaming `_check_fit_data` into  `check_data_shape` and call it here?

Reply:
done. renamed function to `check_input_data_shape`